drafting/conda-template.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+
+# This script reads in a yaml file (qiime2 recipe)
+# Pulls the needed data from that file
+# And uses that data to write to a secondary yaml file
+# Which will be the end result: a conda recipe
+
+import yaml
+import os
+import sys
+from jinja2 import Template
+
+def validate_params(qiime2_recipe, filepath):
```

Comment:
I think you might've taken this one a step too far - this function won't actually even run if a user forgets to provide an input, since the argument lookups are happening _before_ this function is called ( `qiime2_recipe = sys.argv[1]` and `filepath = sys.argv[2]` down below).

You could refactor to something like this, though:

```python
def validate_params(cli_args):
    try:
        qiime2_recipe = cli_args[1]
    ...
    return qiime2_recipe, output_filepath


if __name__ == '__main__':
    input_fp, output_fp = validate_params(sys.argv)
    ...
```

Reply:
@thermokarst thank you, this is really helpful! 🥳